langs/go/run-go

```diff
@@ -0,0 +1,9 @@
+#!/bin/sh -e
+
+# Separate compile and link relies on environment variables less than "go run" does.
+# "go run" also doesn't support reading the file from stdin.
+cat - > /tmp/code.go
+/usr/local/go/bin/go tool compile -o /tmp/code.o /tmp/code.go
+/usr/local/go/bin/go tool link -o /tmp/code /tmp/code.o
```

Comment:
Would `go build` work and make this simpler?

Reply:
I tried it and it complained that a bunch of environment variables were not set. This seemed a bit simpler in that there was less environment setup.

Reply:
I got the same errors for `go build` and `go run`.

Reply:
ah i see, fair enough, never used the `tool` subcommand before. I'll give this a play in a bit, its looks really good, thanks!

Reply:
The error was `build cache is required, but could not be located: GOCACHE is not defined and neither $XDG_CACHE_HOME nor $HOME are defined`.